OpenEdXMobile/src/main/java/org/edx/mobile/course/CourseAPI.kt

```diff
@@ -0,0 +1,324 @@
+package org.edx.mobile.course
+
+import android.content.Context
+import okhttp3.ResponseBody
+import org.edx.mobile.event.EnrolledInCourseEvent
+import org.edx.mobile.exception.CourseContentNotValidException
+import org.edx.mobile.http.callback.ErrorHandlingCallback
+import org.edx.mobile.http.constants.ApiConstants
+import org.edx.mobile.http.constants.TimeInterval
+import org.edx.mobile.http.notifications.ErrorNotification
+import org.edx.mobile.http.notifications.SnackbarErrorNotification
+import org.edx.mobile.http.util.CallUtil
+import org.edx.mobile.interfaces.RefreshListener
+import org.edx.mobile.model.Page
+import org.edx.mobile.model.api.CourseComponentStatusResponse
+import org.edx.mobile.model.api.CourseUpgradeResponse
+import org.edx.mobile.model.api.EnrolledCoursesResponse
+import org.edx.mobile.model.api.EnrollmentResponse
+import org.edx.mobile.model.course.BlockModel
+import org.edx.mobile.model.course.BlockType
+import org.edx.mobile.model.course.BlocksCompletionBody
+import org.edx.mobile.model.course.CourseBannerInfoModel
+import org.edx.mobile.model.course.CourseComponent
+import org.edx.mobile.model.course.CourseDates
+import org.edx.mobile.model.course.CourseDetail
+import org.edx.mobile.model.course.CourseStatus
+import org.edx.mobile.model.course.CourseStructureV1Model
+import org.edx.mobile.model.course.DiscussionBlockModel
+import org.edx.mobile.model.course.DiscussionData
+import org.edx.mobile.model.course.EnrollBody
+import org.edx.mobile.model.course.HtmlBlockModel
+import org.edx.mobile.model.course.IBlock
+import org.edx.mobile.model.course.ResetCourseDates
+import org.edx.mobile.model.course.VideoBlockModel
+import org.edx.mobile.model.course.VideoData
+import org.edx.mobile.module.prefs.UserPrefs
+import org.edx.mobile.util.Config
+import org.edx.mobile.view.common.TaskProgressCallback
+import org.greenrobot.eventbus.EventBus
+import org.json.JSONObject
+import retrofit2.Call
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class CourseAPI @Inject constructor(
+    private var config: Config,
+    private val courseService: CourseService,
+    private val userPrefs: UserPrefs
+) {
+    /**
+     * @return Enrolled courses of given user.
```

Comment:
As we are accessing AppConfig as well in this API, so better to update the doc accordingly.